python/lsst/ip/isr/deferredCharge.py

```diff
@@ -0,0 +1,639 @@
+            return False
+        if self.coeffs != other.coeffs:
+            return False
+        return True
+
+    @property
+    def trap_array(self):
+        return self._trap_array
+
+    @property
+    def trapped_charge(self):
+        return self._trapped_charge
+
+    def initialize(self, ny, nx, prescan_width):
+        """Initialize trapping arrays for simulated readout.
+
+        Parameters
+        ----------
+        ny : `int`
+            Number of rows to simulate.
+        nx : `int`
+            Number of columns to simulate.
+        prescan_width : `int`
+            Additional transfers due to prescan.
+        """
+        if self.pixel > nx+prescan_width:
+            raise ValueError('Trap location {0} must be less than {1}'.format(self.pixel,
+                                                                              nx+prescan_width))
+
+        self._trap_array = np.zeros((ny, nx+prescan_width))
+        self._trap_array[:, self.pixel] = self.size
+        self._trapped_charge = np.zeros((ny, nx+prescan_width))
+
+    def release_charge(self):
+        """Release charge through exponential decay.
+
+        Returns
+        -------
+        released_charge : `float`
+            Charge released.
+        """
+        released_charge = self._trapped_charge*(1-np.exp(-1./self.emission_time))
+        self._trapped_charge -= released_charge
+
+        return released_charge
+
+    def trap_charge(self, free_charge):
+        """Perform charge capture using a logistic function.
+
+        Parameters
+        ----------
+        free_charge : `float`
+            Charge available to be trapped.
+
+        Returns
+        -------
+        captured_charge : `float`
+            Amount of charge actually trapped.
+        """
+        captured_charge = (np.clip(self.capture(free_charge), self.trapped_charge, self._trap_array)
+                           - self.trapped_charge)
+        self._trapped_charge += captured_charge
+
+        return captured_charge
+
+    def capture(self, pixel_signals):
+        """Trap capture function.
+
+        Parameters
+        ----------
+        pixel_signals : `list` [`float`]
+            Input pixel values.
+
+        Returns
+        -------
+        captured_charge : `list` [`float`]
+            Amount of charge captured from each pixel.
+        """
+        if self.trap_type == 'linear':
+            scaling = self.coeffs[0]
+            return np.minimum(self.size, pixel_signals*scaling)
+        elif self.trap_type == 'logistic':
+            f0, k = (self.coeffs[0], self.coeffs[1])
+            return self.size/(1.+np.exp(-k*(pixel_signals-f0)))
+        elif self.trap_type == 'spline':
+            return self.interp(pixel_signals)
+
+
+class DeferredChargeCalib(IsrCalib):
+    """Calibration containing deferred charge/CTI parameters.
+
+    Parameters
+    ----------
+    detector : `lsst.afw.cameraGeom.Detector`, optional
```

Comment:
Removed here, because the `super().__init__(**kwargs)` should pass `detector` in a way that populates its values into the calib (and removed in the `__init__` definition).